src/zenml/steps/utils.py

```diff
@@ -411,6 +411,9 @@ def Do(
         if return_type is not None:
             if isinstance(return_type, Output):
                 # Resolve named (and multi-) outputs.
+                # breakpoint()
+                if sum(1 for dummy_var in return_type.items()) == 1:
```

Comment:
Is there a reason len(return_type.items()) == 1 does not work here?

Reply:
You can't call `len()` on a generator.

Reply:
Totally makes sense, wasn't aware its a generator :)